src/orion/algo/ax/axoptimizer.py

```
@@ -112,3 +112,47 @@ def is_done(self):
         """Return True, if an algorithm holds that there can be no further improvement."""
         # NOTE: Drop if base implementation is fine.
         return super(AxOptimizer, self).is_done
+
+def orion_space_to_axoptimizer_space(orion_space):
+    """Convert Orion's definition of problem's domain to an axoptimizer
+    compatible one."""
+    axoptimizer_space = []
+
+    for key, dimension in orion_space.items():
+        dimensions_dict = {}
+        dimensions_dict["name"] = dimension.name
+
+        if (dimension.prior_name == 'choices'):
+            dimensions_dict["type"] = "choice"
+            dimensions_dict["values"] = dimension.categories
+            axoptimizer_space.append(dimensions_dict)
+
+        else:
+            dimensions_dict["type"] = "range"
+            dimensions_dict["bounds"] = dimension.interval()
+
+            if (dimension.prior_name == 'uniform'):
+                dimensions_dict["value_type"] = "float"
+                dimensions_dict["log_scale"] = False
+
+            if (dimension.prior_name == 'reciprocal'):
+                dimensions_dict["value_type"] = "float"
+                dimensions_dict["log_scale"] = True
+
+            if (dimension.prior_name == 'int_uniform'):
+                dimensions_dict["value_type"] = "int"
+                dimensions_dict["log_scale"] = False
+
+            if (dimension.prior_name == 'int_reciprocal'):
+                dimensions_dict["value_type"] = "int"
+                dimensions_dict["log_scale"] = True
```

```suggestion
            dimensions_dict["log_scale"] = dimension.prior_name.endswith('reciprocal')
            if dimension.prior_name.startswith('int'):
                dimensions_dict["value_type"] = "int"
            else:
                dimensions_dict["value_type"] = "float"
```

---

Lot less verbose that is true but I also like the `TypeError`. If there are no plans to add more types and since this doesn't involves that much code overall, I would have even be tempted into having 4 `if`s for the 4 types, with duplicated codes for the creation of `dict`s, followed by a `else: raise` as this would allow to quickly see what produces what instead of have you brain work the logic. At least for now until a change in one of Oríon or Ax forces a refactoring.

But I'm ok with all ways and if the main maintainer prefers less code that is all good and the question here is more out of curiosity if there are plans to have more of types (maybe `complex_reciprocal`)?

---

There is no plan for complex_reciprocal but we could have lognormal at some point, so we would need to revert to a switch-case approach like here. I'm fine with ifs too. :)